lib/internal/Magento/Framework/Filter/Template.php

```diff
@@ -371,4 +378,72 @@ protected function getStackArgs($stack)
         }
         return $stack;
     }
+
+    /**
+     * Filter the string as template.
+     *
+     * @param string $value
+     * @example syntax {{for item in order.all_visible_items}} sku: {{var item.sku}}<br>name: {{var item.name}}<br> {{/for}} order items collection.
+     * @example syntax {{for thing in things}} {{var thing.whatever}} {{/for}} e.g.:custom collection.
+     * @return string
+     */
+    protected function filterFor($value)
+    {
+        if (preg_match_all(self::LOOP_PATTERN, $value, $constructions, PREG_SET_ORDER)) {
+            foreach ($constructions as $construction) {
+
+                if (sizeof($construction) < 5) {
+                    return $value;
+                }
+
+                $full_text_to_replace = $construction[0];
+                $objectArrayData = $this->getVariable($construction[3], '');
+                $loop_text_to_replace = $construction[4];
+                $vName = preg_replace('/\s+/', '', $construction[1]);
+
+                if (is_array($objectArrayData) || $objectArrayData instanceof \Traversable) {
+
+                    $loopText = [];
+                    $indexCount = 1;
+                    $loop = new \Magento\Framework\DataObject;
+
+                    foreach ($objectArrayData as $k => $objectData) {
+
+                        if (!$objectData instanceof \Magento\Framework\DataObject) { // is array?
+
+                            if (!is_array($objectData)) {
+                                continue;
+                            }
+
+                            $_item = new \Magento\Framework\DataObject;
+                            $_item->setData($objectData);
+                            $objectData = $_item;
+                        }
+
+                        $loop->setData('index', $indexCount++);
+                        $loop->setData('loop');
+                        $this->templateVars['loop'] = $loop;
```

Comment:
In order the the the index value. e.g.: {{var loop.index}}. Line 424 will be removed.